### ISSUE TITLE
MethodAnnotationsScanner is not getting configured unless there is at least one class using the annotation

### ISSUE THREAD

**Issue by sivaprasadreddy** [ronmamo/reflections#297]
I am using reflections:0.9.12 and I want to scan for Constructors which have some annotation, let's say `@Autowired`.

```
public class UserService {
	private UserRepository repository;

	public UserService() {
	}

	@Autowired
	public UserService(UserRepository repository) {
		this.repository = repository;
	}
}
```

```
Reflections reflections = new Reflections(new ConfigurationBuilder()
		.setUrls(ClasspathHelper.forPackage("com.sivalabs.myapp"))
		.setScanners(new MethodAnnotationsScanner()));

Set<Constructor> constructorsAnnotatedWith = reflections.getConstructorsAnnotatedWith(Autowired.class);
```

This code is working fine if there is at least one class using `@Autowired` annotation. But if there is no class using `@Autowired` annotation in the package `com.sivalabs.myapp` then following exception is being thrown:

```
Exception in thread "main" org.reflections.ReflectionsException: Scanner MethodAnnotationsScanner was not configured
	at org.reflections.Store.get(Store.java:39)
	at org.reflections.Store.get(Store.java:61)
	at org.reflections.Store.get(Store.java:46)
	at org.reflections.Reflections.getConstructorsAnnotatedWith(Reflections.java:515)
```

**Expectation:** `MethodAnnotationsScanner` should be registered even when there are no classes using any particular annotation and when `reflections.getConstructorsAnnotatedWith()` is called it should return empty result.

**Comment by dota17** [comment]
I wrote a test case according to your description and tried it, it can work without any exception. But when I change the parameter of the function `forPackage()` to wrong, the error message is the same as yours. So I think your function parameters may be wrong.

**Comment by sivaprasadreddy** [comment]
The package I used **("com.sivalabs.myapp" )** is correct and I have classes exists in that package also. But even if the package name given for `forPackage()` doesn't exists, it should return empty Set instead of throwing exception.

**Comment by dota17** [comment]
Yes, this exception will occurs when annotations are not used at all.  If I use other annotations in another class, this exception will not occur. I will try to figure this out.

**Comment by dota17** [comment]
In reflections:0.9.12, if your scanner can't scan the class, method, annotation and so on  you want, nothing will be placed in the  storeMap. As a result, when reflections.getConstructorsAnnotatedWith() is called, an expection will be thrown here: 
https://github.com/ronmamo/reflections/blob/9a1ac08d59311a91f3dad06f0c831eccdc47f1e3/src/main/java/org/reflections/Store.java#L43

But this problem has been fixed in the latest master branch. When the class Store is initialized，storeMap will get an initial value。See:
https://github.com/ronmamo/reflections/blob/9a1ac08d59311a91f3dad06f0c831eccdc47f1e3/src/main/java/org/reflections/Store.java#L28

**Comment by veselov** [comment]
Bump. Is it possible to make a release with this fix?

**Comment by ronmamo** [comment]
apologize for this inconvenience !

scanner was not configured exception - as @dota17 mentioned, this is a known issue in 0.9.12, a simple workaround is to check if the getStore() contains index for the scanner before querying. [next version 0.10](https://github.com/ronmamo/reflections/pull/335) fixes this.

**Comment by ronmamo** [comment]
fixed on 0.10
